### DISCUSSION TITLE
Minimum Number of Data Points for AutoTS

### DISCUSSION THREAD

**Discussion by Marius-Mors** [winedarksea/AutoTS#229] [Q&A]
Hello everyone,
I have a question regarding the minimum number of data points required for AutoTS to make forecasts. Is there a specific rule indicating how many data points are needed at a minimum to make predictions for 1, 2, 3, 4, ... months? I am interested in creating forecasts for 12 and 24 months and would like to know how many months of data AutoTS needs for this purpose.

Thank you in advance!

**Comment by winedarksea** [comment]
no specific rule...

Generally AutoTS needs a decent bit of data for cross validation, because it needs to do holdouts to figure out which model is best. Although you can set forecast_length smaller in AutoTS() then in .predict make it larger if your data is limited. So you could set AutoTS(forecast_length=12) for model selection then `model.predict(forecast_length=24) to get the max.

There's also an arg `min_allowed_train_percent` which is there to complain if your data is too short.

My general feeling is that 2 years of data** would be enough to get started with (only 1 evaluation holdout, so num_validations=0) and that you would see model performance improve the more data you add until you got up to about 5 years of history, after which point additional history won't usually have much added value (and sometimes actually makes it worse, I've seen this in cases where old data was on no longer relevant market conditions). 

** on weekly/monthly data

generally a rule might be 3x your forecast_length desired, for more reasonable performance

It depends on how much variability happens in the data. You could predict a flat line with very little data. However much seasonality exists also matters. You would generally want 3x repetitions of a seasonality that is important. And holidays are particularly tricky, as they can look like anomalies and move around sometimes, it takes _at least_ 3x repetitions to start picking up patterns in holidays, usually more.

if you really have a tiny amount of data not using AutoTS but using something like a simple linear regression or naive is about all you can do.
